test/http2_test/test_goaway.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import time
+
+import http2_base_server
+
+class TestcaseGoaway(object):
+  """ 
+    This test does the following:
+      Process incoming request normally, i.e. send headers, data and trailers.
+      Then send a GOAWAY frame with the stream id of the processed request.
+      It checks that the next request is made on a different TCP connection.
+  """
+  def __init__(self, iteration):
+    self._base_server = http2_base_server.H2ProtocolBaseServer()
+    self._base_server._handlers['RequestReceived'] = self.on_request_received
+    self._base_server._handlers['DataReceived'] = self.on_data_received
+    self._base_server._handlers['SendDone'] = self.on_send_done
+    self._base_server._handlers['ConnectionLost'] = self.on_connection_lost
+    self._ready_to_send = False
+    self._iteration = iteration
+
+  def get_base_server(self):
+    return self._base_server
+
+  def on_connection_lost(self, reason):
+    logging.info('Disconnect received. Count %d' % self._iteration)
+    # _iteration == 2 => Two different connections have been used.
+    if self._iteration == 2:
```

Comment:
After looking at the base server implementation, I think I understand what this accomplishes: if _iteration != 2, the server won't acknowledge the connection lost, and consequently the server won't be shutdown (which happens from _base_server.on_connection_lost()). This would then be regarded as a failed test.  

Would it be better to "fail fast" here and throw an exception if _iteration != 2? Otherwise, it seems like the test runner would have to wait until the test timeouts to see that the server is still running and the test failed, if I understand how the test driver will execute.